cocotb/types/logic_array.py

```diff
@@ -189,6 +201,31 @@ def is_resolvable(self) -> bool:
 
     @property
     def integer(self) -> int:
+        """
+        Raises ValueError if the values cannot be converted to binary ('X' or 'Z')
+        """
+        return int(self)
+
+    def __eq__(self, other):
+        """
+        Overrides the default implementation to enable comparision
+        with integers and strings (case insensitive)
+        """
+        if isinstance(other, int):
+            return self.integer == other
```

Comment:
Lets not make an assumption about binary representation.
```suggestion
            return self == LogicArray(other)
```